java/core/src/main/java/co/worklytics/psoxy/PsoxyModule.java

```diff
@@ -73,13 +75,28 @@ static SourceAuthStrategy sourceAuthStrategy(ConfigService configService, Set<So
     @Provides
     static OAuthRefreshTokenSourceAuthStrategy.TokenRequestPayloadBuilder tokenRequestPayloadBuilder(ConfigService configService,
                                                                                                      Set<OAuthRefreshTokenSourceAuthStrategy.TokenRequestPayloadBuilder> payloadBuilders) {
-        String identifier =
-            configService.getConfigPropertyOrError(OAuthRefreshTokenSourceAuthStrategy.ConfigProperty.GRANT_TYPE);
-        return payloadBuilders
-            .stream()
-            .filter(impl -> Objects.equals(identifier, impl.getGrantType()))
-            .findFirst()
-            .orElseThrow(() -> new Error("No TokenRequestPayloadBuilder impl supporting oauth grant type: " + identifier));
+        // Final instantiation of configs are per-function. Grant type is dependent of type of auth
+        // strategy so might not exist for certain functions.
+        // If it is mis-configured will throw NPE at some point
+        Optional<String> grantTypeOptional = configService.getConfigPropertyAsOptional(OAuthRefreshTokenSourceAuthStrategy.ConfigProperty.GRANT_TYPE);
```

Comment:
change `getConfigPropertyOrError`, if function does not use GRANT_TYPE shouldn't err.